src/screens/tabs/Profile/screens/CreateEditAccount/index.js

```diff
@@ -9,33 +10,56 @@ import UserIcon from '@/commonComponents/UserIcon';
 import RainbowButton from '@/components/buttons/RainbowButton';
 import Text from '@/components/common/Text';
 import { FontStyles } from '@/constants/theme';
-import useAccounts from '@/hooks/useAccounts';
+import {
+  createSubaccount,
+  editSubaccount,
+  importAccountFromPem,
+} from '@/redux/slices/keyring';
 
 import EditEmoji from '../EditEmoji';
 import styles from './styles';
 
-const CreateEditAccount = ({ modalRef, account, accountsModalRef }) => {
+/**
+ * @param {{ modalRef: any, accountsModalRef?: any, account?: any, pem?: string, createImportModalRef?: any } param
+ */
+const CreateEditAccount = ({
+  modalRef,
+  account,
+  accountsModalRef,
+  pem,
+  createImportModalRef,
+}) => {
   const { t } = useTranslation();
   const editEmojiRef = useRef(null);
   const [accountName, setAccountName] = useState('');
   const [editTouched, setEditTouched] = useState(false);
   const [emoji, setEmoji] = useState('');
+  const dispatch = useDispatch();
 
-  const { onCreate, onEdit } = useAccounts();
+  const nameAndIcon = {
+    name: accountName,
+    icon: emoji,
+  };
 
   const onPress = () => {
-    account
-      ? onEdit({
-          walletId: account.walletId,
-          name: accountName,
-          icon: emoji,
-        })
-      : onCreate({
-          name: accountName,
-          icon: emoji,
-        });
+    dispatch(
+      account
+        ? editSubaccount({
+            walletId: account.walletId,
+            ...nameAndIcon,
+          })
+        : pem
+        ? importAccountFromPem({
+            ...nameAndIcon,
+            pem,
+          })
+        : createSubaccount(nameAndIcon)
+    );
```

Comment:
i think it'll be more readable if we replace the double ternary with ifs

if(account) -> dispatch
else if(pem) -> dispatch
else -> dispatch